form/src/main/java/org/dhis2/form/ui/validation/validators/FieldMaskValidator.kt

```diff
@@ -1,15 +1,35 @@
 package org.dhis2.form.ui.validation.validators
 
+import java.lang.Exception
+import java.util.regex.Pattern
 import org.dhis2.form.ui.validation.failures.FieldMaskFailure
 import org.hisp.dhis.android.core.arch.helpers.Result
 import org.hisp.dhis.android.core.common.valuetype.validation.validators.ValueTypeValidator
+import timber.log.Timber
 
 class FieldMaskValidator(val fieldMask: String) : ValueTypeValidator<FieldMaskFailure> {
 
     override fun validate(value: String): Result<String, FieldMaskFailure> {
-        return when (value.matches(fieldMask.toRegex())) {
-            true -> Result.Success(value)
-            else -> Result.Failure(FieldMaskFailure.WrongPatternException)
+        return if (value.isEmpty()) {
+            Result.Success(value)
+        } else if (fieldMaskIsCorrect()) {
+            if (value.matches(fieldMask.toRegex())) {
+                Result.Success(value)
+            } else {
+                Result.Failure(FieldMaskFailure.WrongPatternException)
+            }
+        } else {
+            Result.Failure(FieldMaskFailure.InvalidPatternException)
+        }
+    }
+
+    private fun fieldMaskIsCorrect(): Boolean {
+        return try {
+            Pattern.compile(fieldMask)
+            true
+        } catch (e: Exception) {
+            Timber.d(e)
+            false
```

Comment:
@Balcan do you think we could cover it with unit test?

Reply:
There is already a unit test class ;)

Reply:
where? i dont see it :(